server/services/ApiKeyService.js

```diff
@@ -0,0 +1,74 @@
+const FS = require('fs');
+const log = require("loglevel");
+const path = require("path");
+const HttpError = require("../utils/HttpError");
+const ApiKeyRepository = require("../repositories/ApiKeyRepository");
+const expect = require("expect-runtime");
+
+// PRIVATE and PUBLIC key
+const privateKEY = FS.readFileSync(path.resolve(__dirname, '../../config/jwtRS256.key'), 'utf8');
+const publicKEY = FS.readFileSync(path.resolve(__dirname, '../../config/jwtRS256.key.pub'), 'utf8');
+
+
+class ApiKey{
+  constructor(){
+    this.apiKeyRepository = new ApiKeyRepository();
+  }
+
+  async check(apiKey){
+    if (!apiKey) {
+      log.log('ERROR: Invalid access - no API key');
+      throw new HttpError(401,'Invalid access - no API key');
+    }
+    let result;
+    try{
+      result = await this.apiKeyRepository.getByApiKey(apiKey);
+    }catch(e){
+      if(e.code === 404){
+        log.debug("404 -> 401");
+        throw new HttpError(401,'Invalid API access');
+      }else{
+        log.debug("throw e:", e.message);
+        throw e;
+      }
```

Comment:
What's the difference between how we handle HttpErrors compared to throwing an exception?

Reply:
This is the principle of the throw-catch paradigm, if I know how to deal with an exception, then catch it and do stuff, if I don't know, them just simply continue to throw it out.
Here, the call of `getByApiKey` might throw 404, means can not find the key, then I translate it to a more friendly or appropriate response to clients. 
There are maybe something else, let's assume, a DB operation or something else, it will be throw to the top of the error handler, if the error handler can not deal with this type of error too, then, it will translate it to a 500 'Some unknown error occurred'